### DISCUSSION TITLE
How to run different jobs on different agents within the same code location in hybrid deployment?

### DISCUSSION THREAD

**Discussion by polivbr** [dagster-io/dagster#21490] [Q&A]
Within a single pipeline, we have a need for certain jobs to be run within specific environments due to platform or license restrictions. While we can assign an agent queue to the code location, it does not appear that there's a way to specify the queue at the asset/job/op level. If we were running self-hosted we could potentially utilize celery and target specific celery queues, but this is not an option in Dagster+.

Is there any way to accomplish this?

**Comment by garethbrickman** [comment]
It seems like this is not possible in Dagster+ at this time. The documentation for [Routing requests to specific agents](https://docs.dagster.io/dagster-plus/deployment/agents/running-multiple-agents#routing-requests-to-specific-agents) indicates that only entire code locations can be routed to specific agents.
